### PR TITLE
Add javadoc to pull request status

### PR DESCRIPTION
**Pull request by Kostronor**

### CODE REVIEW

**Comment by Kostronor** [comment]
:shipit: @cltnschlosser @Raycoms I fixed the javadoc mess :D

**Comment by Raycoms** [comment]
:shipit: nice

**Comment by cltnschlosser** [comment]
So what exactly does this do? Like how does it work?

File: javadoc.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+echo "Javadoc-testing commit $TRAVIS_COMMIT"
```
Comment (cltnschlosser) [inline javadoc.sh]: Debug line?

File: javadoc.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+if [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ -n "${GITHUB_TOKEN:-}" ]; then
```
Comment (cltnschlosser) [inline javadoc.sh]: Why aren't we doing this on pull request?

File: javadoc.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            --request POST \
+            --data '{"state":"pending",
+            "target_url":"https://travis-ci.org/Minecolonies/minecolonies/builds/{$TRAVIS_BUILD_ID}",
+            "description":"Javadoc build in progress...",
+            "context":"javadoc"}' \
+            "https://api.github.com/repos/Minecolonies/minecolonies/statuses/$TRAVIS_COMMIT"
+            
+      echo "Start javadoc gradle..."
+      
+      if ./gradlew javadoc; then
```
Comment (cltnschlosser) [inline javadoc.sh]: So, this allows us to create a custom PR Acceptance criteria that we change using api calls?

File: javadoc.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            curl -H "Content-Type: application/json" -H "Authorization: token $GITHUB_TOKEN" \
```
Comment (cltnschlosser) [inline javadoc.sh]: Also, been curious, is this the Minecolonies organization's token?

File: javadoc.sh (hunk 1 above)
Comment (Kostronor) [inline javadoc.sh]: Some nice output to make it look cleaner

File: javadoc.sh (hunk 2 above)
Comment (Kostronor) [inline javadoc.sh]: Because then we would report the check two times. A pull request always has the push build.

File: javadoc.sh (hunk 3 above)
Comment (Kostronor) [inline javadoc.sh]: Yep, exactly

File: javadoc.sh (hunk 4 above)
Comment (Kostronor) [inline javadoc.sh]: This is the MinecoloniesSonar user's token. Someone I created so no one of us had to give the token.

**Pull request merged**